lib/java_buildpack/framework/introscope_agent.rb

```diff
@@ -44,6 +44,7 @@ def release
           .add_system_property('com.wily.introscope.agent.agentName', agent_name(credentials))
           .add_system_property('introscope.agent.defaultProcessName', default_process_name)
 
+        java_opts.add_system_property('agentManager.credential', credential(credentials)) if credential(credentials)
```

Comment:
Yep.  And if the Ruby VM has any smarts at all, it'll optimize that out to a single `credential(credentials)` call.